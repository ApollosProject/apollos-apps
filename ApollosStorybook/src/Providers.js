import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
