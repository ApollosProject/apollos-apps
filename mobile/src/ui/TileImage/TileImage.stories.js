import React from 'react';
import { storiesOf } from '@storybook/react-native';

import TileImage from './';

storiesOf('TileImage', module)
  .add('Default', () => (
    <TileImage
      image={'https://picsum.photos/600/400/?random'}
      link={'https://github.com'}
      text={'So cool!'}
    />
  ))
  .add('without text', () => (
    <TileImage
      image={'https://picsum.photos/600/400/?random'}
      link={'https://github.com'}
    />
  ));
