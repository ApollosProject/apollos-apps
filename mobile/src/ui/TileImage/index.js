import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';
import PropTypes from 'prop-types';
import { pure, compose } from 'recompose';

import { H6 } from 'ui/typography';
import styled from 'ui/styled';
import { withTheme } from 'ui/theme';
import GradientOverlayImage from 'ui/GradientOverlayImage';

const CardView = styled(
  ({ theme }) => ({
    borderRadius: theme.sizing.borderRadius,
    overflow: 'hidden',
  }),
  'TileImage'
)(View);

const Title = styled(
  ({ theme }) => ({
    position: 'absolute',
    bottom: theme.sizing.baseUnit,
    left: theme.sizing.baseUnit,
    backgroundColor: theme.colors.transparent,
    color: theme.colors.lightPrimary,
  }),
  'TileImage.Text'
)(H6);

const enhance = compose(
  withTheme(({ theme: { colors } = {} } = {}) => ({ theme: { colors } })),
  pure
);

const TileImage = enhance(({ image, link, onPressItem, text, theme }) => (
  <TouchableWithoutFeedback onPress={() => onPressItem({ ...link })}>
    <CardView>
      <GradientOverlayImage
        source={image}
        overlayColor={text ? theme.colors.black : null}
      />
      {text ? <Title>{text}</Title> : null}
    </CardView>
  </TouchableWithoutFeedback>
));

TileImage.propTypes = {
  image: GradientOverlayImage.propTypes.source,
  link: PropTypes.string.isRequired,
  onPressItem: PropTypes.func,
  text: PropTypes.string,
};

export default TileImage;
