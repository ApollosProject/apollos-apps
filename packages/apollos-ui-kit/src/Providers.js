import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
