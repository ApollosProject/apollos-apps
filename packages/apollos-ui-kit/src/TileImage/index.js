import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { pure, compose } from 'recompose';

import { H4 } from '../typography';
import styled from '../styled';
import { withTheme } from '../theme';
import GradientOverlayImage from '../GradientOverlayImage';
import TouchableScale from '../TouchableScale';

const CardView = styled(
  ({ theme }) => ({
    borderRadius: theme.sizing.baseUnit,
    overflow: 'hidden',
    width: '100%',
    height: '100%',
    aspectRatio: 1,
  }),
  'TileImage'
)(View);

const Title = styled(
  ({ theme }) => ({
    position: 'absolute',
    bottom: theme.sizing.baseUnit,
    left: theme.sizing.baseUnit,
    right: theme.sizing.baseUnit,
    backgroundColor: theme.colors.transparent,
    color: theme.colors.lightPrimary,
  }),
  'TileImage.Text'
)(H4);

const SquareGradientOverlayImage = styled({
  aspectRatio: 1,
})(GradientOverlayImage);

const enhance = compose(
  withTheme(({ theme: { colors } = {} } = {}) => ({ theme: { colors } })),
  pure
);

const TileImage = enhance(
  ({ image, link, onPressItem, text, theme, isLoading }) => (
    <TouchableScale onPress={() => !isLoading && onPressItem({ ...link })}>
      <CardView>
        <SquareGradientOverlayImage
          source={image}
          isLoading={isLoading}
          maintainAspectRatio={false}
          overlayColor={text ? theme.colors.black : null}
        />
        <Title isLoading={isLoading}>{text}</Title>
      </CardView>
    </TouchableScale>
  )
);

TileImage.propTypes = {
  image: GradientOverlayImage.propTypes.source,
  link: PropTypes.string,
  onPressItem: PropTypes.func,
  text: PropTypes.string,
  isLoading: PropTypes.bool,
};

TileImage.defaultProps = {
  text: '',
};

export default TileImage;
