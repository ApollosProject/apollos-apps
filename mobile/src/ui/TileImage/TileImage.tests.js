import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'TestProviders';

import TileImage from './';

describe('the TileImage component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <TileImage
          image={'https://picsum.photos/600/400/?random'}
          link={'https://github.com'}
          text={'So cool!'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render without text', () => {
    const tree = renderer.create(
      <Providers>
        <TileImage
          image={'https://picsum.photos/600/400/?random'}
          link={'https://github.com'}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('passes an onPressItem prop', () => {
    const tree = renderer.create(
      <Providers>
        <TileImage
          image={'https://picsum.photos/600/400/?random'}
          link={'https://github.com'}
          onPressItem={jest.fn}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
