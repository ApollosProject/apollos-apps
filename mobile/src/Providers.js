import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
