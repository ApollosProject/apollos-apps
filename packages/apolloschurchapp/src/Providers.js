import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
