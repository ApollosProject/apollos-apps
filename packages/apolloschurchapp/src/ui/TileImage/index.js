import React from 'react';
import { TouchableWithoutFeedback, View } from 'react-native';
import PropTypes from 'prop-types';
import { pure, compose } from 'recompose';

import { H4 } from 'apolloschurchapp/src/ui/typography';
import styled from 'apolloschurchapp/src/ui/styled';
import { withTheme } from 'apolloschurchapp/src/ui/theme';
import GradientOverlayImage from 'apolloschurchapp/src/ui/GradientOverlayImage';

const CardView = styled(
  ({ theme }) => ({
    borderRadius: theme.sizing.borderRadius,
    overflow: 'hidden',
    width: '100%',
    height: '100%',
    aspectRatio: 1,
  }),
  'TileImage'
)(View);

const Title = styled(
  ({ theme }) => ({
    position: 'absolute',
    bottom: theme.sizing.baseUnit,
    left: theme.sizing.baseUnit,
    right: theme.sizing.baseUnit,
    backgroundColor: theme.colors.transparent,
    color: theme.colors.lightPrimary,
  }),
  'TileImage.Text'
)(H4);

const SquareGradientOverlayImage = styled({
  aspectRatio: 1,
})(GradientOverlayImage);

const enhance = compose(
  withTheme(({ theme: { colors } = {} } = {}) => ({ theme: { colors } })),
  pure
);

const TileImage = enhance(
  ({ image, link, onPressItem, text, theme, isLoading }) => (
    <TouchableWithoutFeedback
      onPress={() => !isLoading && onPressItem({ ...link })}
    >
      <CardView>
        <SquareGradientOverlayImage
          source={image}
          isLoading={isLoading}
          maintainAspectRatio={false}
          overlayColor={text ? theme.colors.black : null}
        />
        <Title isLoading={isLoading}>{text}</Title>
      </CardView>
    </TouchableWithoutFeedback>
  )
);

TileImage.propTypes = {
  image: GradientOverlayImage.propTypes.source,
  link: PropTypes.string,
  onPressItem: PropTypes.func,
  text: PropTypes.string,
  isLoading: PropTypes.bool,
};

TileImage.defaultProps = {
  text: '',
};

export default TileImage;
