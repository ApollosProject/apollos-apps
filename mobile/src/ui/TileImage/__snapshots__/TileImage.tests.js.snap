// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the TileImage component passes an onPressItem prop 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "borderRadius": 6,
      "height": "100%",
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#A5A5A5",
        "width": "100%",
      }
    }
  >
    <View
      animate={null}
      customAnimate={null}
      onReady={false}
      style={
        Object {
          "aspectRatio": 1,
          "backgroundColor": "#A5A5A5",
          "width": "100%",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Array [
            Object {
              "uri": "https://picsum.photos/600/400/?random",
              "url": "https://picsum.photos/600/400/?random",
            },
          ]
        }
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#A5A5A5",
            "opacity": 0,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 20,
        "color": "#F8F7F4",
        "fontFamily": "InterUI-Bold",
        "fontSize": 16,
        "left": 20,
        "lineHeight": 23.04,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    
  </Text>
</View>
`;

exports[`the TileImage component should render 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "borderRadius": 6,
      "height": "100%",
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#A5A5A5",
        "width": "100%",
      }
    }
  >
    <View
      animate={null}
      customAnimate={null}
      onReady={false}
      style={
        Object {
          "aspectRatio": 1,
          "backgroundColor": "#A5A5A5",
          "width": "100%",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Array [
            Object {
              "uri": "https://picsum.photos/600/400/?random",
              "url": "https://picsum.photos/600/400/?random",
            },
          ]
        }
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#A5A5A5",
            "opacity": 0,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      />
    </View>
    <BVLinearGradient
      colors={
        Array [
          0,
          4278190080,
        ]
      }
      endPoint={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      locations={
        Array [
          0.3,
          1,
        ]
      }
      startPoint={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 20,
        "color": "#F8F7F4",
        "fontFamily": "InterUI-Bold",
        "fontSize": 16,
        "left": 20,
        "lineHeight": 23.04,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    So cool!
  </Text>
</View>
`;

exports[`the TileImage component should render a loading state 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "borderRadius": 6,
      "height": "100%",
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#A5A5A5",
        "width": "100%",
      }
    }
  >
    <View
      animate={null}
      customAnimate={null}
      onReady={false}
      style={
        Object {
          "aspectRatio": 1,
          "backgroundColor": "#A5A5A5",
          "width": "100%",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Array [
            Object {
              "uri": "https://picsum.photos/600/400/?random",
              "url": "https://picsum.photos/600/400/?random",
            },
          ]
        }
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#A5A5A5",
            "opacity": 0,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#A5A5A5",
        "borderRadius": 6,
        "height": 16,
        "marginVertical": 3.5199999999999996,
        "width": "80%",
      }
    }
  />
</View>
`;

exports[`the TileImage component should render without text 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "aspectRatio": 1,
      "borderRadius": 6,
      "height": "100%",
      "overflow": "hidden",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#A5A5A5",
        "width": "100%",
      }
    }
  >
    <View
      animate={null}
      customAnimate={null}
      onReady={false}
      style={
        Object {
          "aspectRatio": 1,
          "backgroundColor": "#A5A5A5",
          "width": "100%",
        }
      }
    >
      <Image
        onLoad={[Function]}
        source={
          Array [
            Object {
              "uri": "https://picsum.photos/600/400/?random",
              "url": "https://picsum.photos/600/400/?random",
            },
          ]
        }
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#A5A5A5",
            "opacity": 0,
            "resizeMode": "cover",
            "width": "100%",
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "bottom": 20,
        "color": "#F8F7F4",
        "fontFamily": "InterUI-Bold",
        "fontSize": 16,
        "left": 20,
        "lineHeight": 23.04,
        "position": "absolute",
        "right": 20,
      }
    }
  >
    
  </Text>
</View>
`;
